### PR TITLE
Support custom DNS alt hosts

### PR DESCRIPTION
**Pull request by nmittler**
This is necessary to support the Kubernetes MCS `clusterset.local` host. This is only preliminary work. Future PRs will provide full support for the alias.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[x] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pkg/dns/client/dns_test.go [hunk 1]
```diff
@@ -419,7 +444,14 @@ func initDNS(t test.Failer) *LocalDNSServer {
 	}
 	testAgentDNS.resolvConfServers = []string{srv}
 	testAgentDNS.StartDNS()
-	testAgentDNS.searchNamespaces = []string{"ns1.svc.cluster.local", "svc.cluster.local", "cluster.local"}
+	testAgentDNS.searchNamespaces = []string{
```
Comment (howardjohn) [inline pkg/dns/client/dns_test.go:447]: searchNamespaces comes from k8s configuring /etc/resolv.conf. In real world, how is this going to work?

Comment (nmittler) [reply pkg/dns/client/dns_test.go]: I didn't actually need this change.

**Comment by nmittler** [comment]
@howardjohn PTAL

File: pkg/dns/proto/nds.proto [hunk 2]
```diff
@@ -20,14 +20,24 @@ package istio.networking.nds.v1;
...
+
+        // The k8s service name. Only applies when registry=`Kubernetes`
         string shortname = 3;
+
+        // The k8s namespace for the service. Only applies when registry=`Kubernetes`
         string namespace = 4;
+
+        // List of alternate hosts to map to the IPs.
+        // Only applies when registry=`Kubernetes`
+        repeated string alt_hosts = 5;
```
Comment (howardjohn) [inline pkg/dns/proto/nds.proto:37]: Thinking about compatibility:
* If we have new istiod, we will send the config and agent will silently ignore. No problem
* If we have new proxy (not supported technically), Istiod will not send alt_hosts. No problem
So LGTM

**Review by howardjohn — APPROVED** [review]

File: pkg/dns/proto/nds.proto (hunk 2 above)
Comment (nmittler) [reply pkg/dns/proto/nds.proto:37]: yeah exactly

**Pull request merged**